nasa/harmony
# Harmony 1218

## Jira Issue ID
HARMONY-1218

## Description
Adds an endpoint to generate bucket policies that will allow Harmony to write output into the given bucket/folder.

## Local Test Steps
Will return message saying it only works for AWS deployments, so it has to be tested in the sandbox.
1. deploy this branch to the sandbox
2. set up a bucket in SIT for testing
3. Set up an environment in Insomnia or similar to query your sandbox harmony
4. query the endpoint to generate a bucket policy (change my-staging-bucket to your bucket name)
{{ _.root }}/staging-bucket-policy?bucketPath=my-staging-bucket
5. Edit the bucket policy on your staging bucket from step 2 and replace it with the generated one
6. Issue the following query specifying the staging bucket for your output
{{ _.root }}/C1234208438-POCLOUD/ogc-api-coverages/1.0.0/collections/bathymetry/coverage/rangeset?concatenate=true&subset=lon(-160%3A160)&subset=lat(-80%3A80)&skipPreview=true&maxResults=3&destinationUrl=s3%3A%2F%2Fmy-staging-bucket
7. Verify that their is a folder in your bucket named after the request ID, and that it has a file called `harmony-jog-status-link` in it as well as another folder named after a work-item ID. The final data file should be in that folder.

OPTIONAL: repeat steps 4-7 using a sub-folder, e.g.,`my-bucket/folder-name` and again using a full S3 url instead of jus `bucket-name/folder-name`, e.g., `s3://my-bucket/folder-name`.

## PR Acceptance Checklist
